src/wallet/wallet2.cpp

```diff
@@ -772,6 +790,60 @@ void wallet2::check_pending_txes()
   }
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::fast_refresh(uint64_t start_height, uint64_t &blocks_start_height, std::list<crypto::hash> &short_chain_history)
+{
+  std::list<crypto::hash> hashes;
+  size_t current_index = m_blockchain.size();
+  while(current_index < m_refresh_from_block_height)
+  {
+    pull_hashes(start_height, blocks_start_height, short_chain_history, hashes);
```

Comment:
Then again, I don't see why it should be broken like this. I think this is actually the same problem with the original refresh <height> - pull_next_blocks() never increments start_height either, so it pulls the same chunk of blocks over and over.